pydrive2/files.py

```diff
@@ -97,6 +98,48 @@ def _GetList(self):
         return result
 
 
+class FakeWriteable(object):
+    def write(self, chunk):
+        self.chunk = chunk
+
+
+class MediaIoReadable(object):
+    def __init__(self, request, chunksize=DEFAULT_CHUNK_SIZE, pre_buffer=True):
+        """File-like wrapper around MediaIoBaseDownload.
+
+    :param pre_buffer: Whether to read one chunk into an internal buffer
+    immediately in order to raise any potential errors.
+    :raises: HttpError
+    """
+        self.done = False
+        self.fd = FakeWriteable()
+        self.downloader = MediaIoBaseDownload(
+            self.fd, request, chunksize=chunksize
+        )
+        self._buffer = None
+        if pre_buffer:
+            self._buffer = self.read()
+
+    def read(self, chunksize=DEFAULT_CHUNK_SIZE):
```

Comment:
it looks like we need to support `encoding` and `mode`

Reply:
how dow `MediaIoBaseDownload` operate btw? always binary mode? does it decode the result somehow?

Reply:
It doesn't do any decoding.

https://googleapis.github.io/google-api-python-client/docs/epy/googleapiclient.http-pysrc.html#MediaIoBaseDownload

basically does:

```python
resp, content = _retry_request(
	request.http, num_retries, "media download", sleep, random, 
	request.uri, "GET", headers)
fd.write(content)
```

Where `chunksize` is injected into `headers`. Perhaps there are encoding-related things we could inject into headers? There's nothing in `MediaIoBaseDownload` itself for encoding.

Reply:
P.S. I've added an `encoding` argument now similar to `GetContentString()`